src/pages/my_page/user_profile/MyProfile.tsx

```diff
@@ -1,46 +1,177 @@
+                  {({ isSubmitting }: FormikProps<CristinPersonFormData>) => {
+                    return (
+                      <Form>
+                        <InputContainerBox sx={{ display: 'flex', flexDirection: 'row' }}>
+                          <Field name={'preferredFirstName'}>
+                            {({ field, meta: { touched, error } }: FieldProps<string>) => (
+                              <TextField
+                                {...field}
+                                id={field.name}
+                                disabled={!editPreferredNames || isSubmitting}
+                                label={t('my_page.my_profile.preferred_first_name')}
+                                size="small"
+                                variant="filled"
+                                error={touched && !!error}
+                                helperText={<ErrorMessage name={field.name} />}
+                              />
+                            )}
+                          </Field>
+                          <Field name={'preferredLastName'}>
+                            {({ field, meta: { touched, error } }: FieldProps<string>) => (
+                              <TextField
+                                {...field}
+                                id={field.name}
+                                disabled={!editPreferredNames || isSubmitting}
+                                label={t('my_page.my_profile.preferred_last_name')}
+                                size="small"
+                                variant="filled"
+                                error={touched && !!error}
+                                helperText={<ErrorMessage name={field.name} />}
+                              />
+                            )}
+                          </Field>
+                          <Box sx={{ display: 'flex', alignItems: 'center' }}>
+                            <Tooltip title={t('common.edit')}>
+                              <IconButton onClick={() => setEditPreferredNames(!editPreferredNames)}>
+                                <EditIcon sx={{ width: '1.2rem' }} />
+                              </IconButton>
+                            </Tooltip>
+                          </Box>
+                        </InputContainerBox>
+                        <Box sx={{ display: 'flex' }}>
+                          <Typography sx={{ mt: '0.6rem' }}>{t('common.orcid')}</Typography>
+                          <UserOrcid user={user} />
+                        </Box>
+                        <Box sx={{ display: 'flex', alignItems: 'center', justifyContent: 'right' }}>
+                          <LoadingButton loading={isSubmitting} variant="contained" type="submit">
+                            {t('common.save')}
+                          </LoadingButton>
+                        </Box>
+                        <Box sx={{ gridArea: 'roles', gridRow: 2 }}>
```

Comment:
gridRow gjentar bare samme plassering som gridArea (?)
```suggestion
                        <Box sx={{ gridArea: 'roles' }}>
```